docs/guide/migrating-to-esm.md

```diff
@@ -0,0 +1,145 @@
+# Migrating to ESM in ApostropheCMS
+
+Starting with version 4.9.0, ApostropheCMS supports ECMAScript Modules (ESM). This guide will help you convert your existing project to use ESM.
+
+## Understanding ESM vs CommonJS
+
+Before beginning the migration, it's important to understand a few key points:
+
+- At the project level, you must choose either CommonJS (CJS) or ESM - they cannot be mixed
+- Node packages can support both ESM and CJS through dual packaging
+- All new ApostropheCMS starter kits and extension packages will use ESM going forward
+
+## Migration Steps
+
+### 1. Update Package Configuration
+
+Add the following to your `package.json`:
+
+```json
+{
+  "type": "module"
+}
+```
+
+### 2. Modify App Configuration
+
+In your `app.js`, update the root configuration:
+
+```javascript
+// Before
+module.exports = {
+  // ... configuration
+};
+
+// After
+export default {
+  root: import.meta,
+  // ... configuration
+};
+```
+
+### 3. Update Import Statements
+
+#### For Node Modules:
+```javascript
+// Before
+const express = require('express');
+
+// After
+import express from 'express';
+```
+
+#### For Local Files:
+```javascript
+// Before
+const myModule = require('./my-module');
+
+// After
+import myModule from './my-module.js';  // Note: .js extension is required
+```
+
+## Common Patterns and Examples
+
+### Module Imports
+
+```javascript
+// Before
+const { apos } = require('@apostrophecms/apostrophe');
+const { existsSync } = require('fs');
```

Comment:
I don't want people thinking `@apostrophecms/apostrophe` is a thing

```suggestion
const { myModule } = require('@my/module');
const { existsSync } = require('fs');
```

Reply:
Changed.